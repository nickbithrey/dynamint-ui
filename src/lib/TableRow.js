import React from 'react';
import PropTypes from 'prop-types';

const TableRow = ({record = []}) => (
	<tr>
		{record.map((r,i) => <td key={i}>{r}</td>)}
	</tr>
);

TableRow.propTypes = {
	record: PropTypes.arrayOf(PropTypes.node)
}

export default TableRow;