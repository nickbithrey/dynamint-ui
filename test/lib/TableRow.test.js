import React from 'react';
import { shallow } from 'enzyme';

import TableRow from '~/lib/TableRow';

describe('<TableRow /> rendering', () => {

	it('can render standard table row with records', () => {
		const records = [
			'col1',
			'col2',
			'col3'
		]
		const wrapper = shallow( <TableRow record={records} /> );
		
		expect(wrapper).toMatchSnapshot();
	});
	
	it('renders table row with no elements', () => {
		const wrapper = shallow( <TableRow record={[]} /> );

		expect(wrapper.find('td')).toHaveLength(0);
	});
		
	it('renders table row with same number of elements in supplied list', () => {
		const records = [
			"",
			"",
			""
		]
		const load = jest.fn(() => []);
		const wrapper = shallow( <TableRow record={records} /> );
		
		expect(wrapper.find('td')).toHaveLength(records.length);
	});
		
	it('can render different data types', () => {
		const time = Date.now();
		const records = [
			"col1",
			1,
			time
		]
		const load = jest.fn(() => []);
		const wrapper = shallow( <TableRow record={records} /> );
		
		const columns = wrapper.find('td');
		expect(columns.at(0).text()).toBe('col1');
		expect(columns.at(1).text()).toBe('1');
		expect(columns.at(2).text()).toBe('' + time);
	});
		
});