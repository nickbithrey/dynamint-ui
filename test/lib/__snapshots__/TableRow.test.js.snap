// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TableRow /> rendering can render standard table row with records 1`] = `
<tr>
  <td
    key="0"
  >
    col1
  </td>
  <td
    key="1"
  >
    col2
  </td>
  <td
    key="2"
  >
    col3
  </td>
</tr>
`;
